securedrop_client/export.py

```diff
@@ -0,0 +1,238 @@
+        '''
+        filename = os.path.basename(filepath)
+        arcname = os.path.join(cls.DISK_EXPORT_DIR, filename)
+        archive.add(filepath, arcname=arcname, recursive=False)
+
+    def _run_usb_test(self, archive_dir: str) -> None:
+        '''
+        Run usb-test.
+
+        Args:
+            archive_dir (str): The path to the directory in which to create the archive.
+
+        Raises:
+            ExportError: Raised if the usb-test does not return a USB_CONNECTED status.
+        '''
+        logger.debug('Running usb-test...')
+
+        archive_path = self._create_archive(archive_dir, self.USB_TEST_FN, self.USB_TEST_METADATA)
+        status = self._export_archive(archive_path)
+        if status != ExportStatus.USB_CONNECTED.value:
+            raise ExportError(status)
+
+    def _run_disk_test(self, archive_dir: str) -> None:
+        '''
+        Run disk-test.
+
+        Args:
+            archive_dir (str): The path to the directory in which to create the archive.
+
+        Raises:
+            ExportError: Raised if the usb-test does not return a DISK_ENCRYPTED status.
+        '''
+        logger.debug('Running disk-test...')
+
+        archive_path = self._create_archive(archive_dir, self.DISK_TEST_FN, self.DISK_TEST_METADATA)
+
+        status = self._export_archive(archive_path)
+        if status != ExportStatus.DISK_ENCRYPTED.value:
+            raise ExportError(status)
+
+    def _run_disk_export(self, archive_dir: str, filepaths: List[str], passphrase: str) -> None:
+        '''
+        Run disk-test.
+
+        Args:
+            archive_dir (str): The path to the directory in which to create the archive.
+
+        Raises:
+            ExportError: Raised if the usb-test does not return a DISK_ENCRYPTED status.
+        '''
+        logger.debug('Exporting to disk...')
+
+        metadata = self.DISK_METADATA.copy()
+        metadata[self.DISK_ENCRYPTION_KEY_NAME] = passphrase
+        archive_path = self._create_archive(archive_dir, self.DISK_FN, metadata, filepaths)
+
+        status = self._export_archive(archive_path)
+        if status:
+            raise ExportError(status)
+
+        logger.debug('Export successful')
+
+    def run_preflight_checks(self) -> None:
+        '''
+        Run preflight checks to verify that the usb device is connected and luks-encrypted.
+        '''
+        if not self.is_qubes:
+            return
+
+        with TemporaryDirectory() as temp_dir:
+            self._run_usb_test(temp_dir)
+            self._run_disk_test(temp_dir)
+
+    def send_file_to_usb_device(self, filepaths: List[str], passphrase: str) -> None:
+        '''
+        Export the file to the luks-encrypted usb disk drive attached to the Export VM.
+
+        Args:
+            filepath: The path of file to export.
+            passphrase: The passphrase to unlock the luks-encrypted usb disk drive.
+        '''
+        if not self.is_qubes:
+            return
```

Comment:
Export should probably only be tested in Qubes since it requires the Export VM, at least for now, and there's already logging inside the methods that are run when actually exporting documents. But perhaps I should remove the is_qubes check from the Export class all together then. That's seems reasonable.